### ISSUE TITLE
2.0b3 continuous scanning even when using with default settings

### ISSUE THREAD

**Issue by rburgst** [yudielcurbelo/react-qr-scanner#33]
Hi
I recently did a yarn upgrade-interactive and for some unknown reason I got the 2.0 version proposed. Anyway, when running it via the following (what I consider to be the default component integration): 
```tsx
  <Scanner
            onResult={onQrCodeReceived}
            onError={error => console.log('scan error', error?.message)}
          />
```
I do get continuous scanning even though I no longer show a QR code to the camera (so I hear a loud beep every 1s or so).

**Comment by vanthao-github** [comment]
Same here! Please teach me how to fix it?

**Comment by ohager** [comment]
I have similar issues: the Scanner continues scanning even when component got unmounted (routing away). 
Seems that at https://github.com/yudielcurbelo/react-qr-scanner/blob/dev/src/components/Scanner.tsx#L43
an additional `useEffect` is required?

```ts
useEffect(() => {
 return stopScanning;
}, [])
```
something like that

**Comment by ohager** [comment]
> Same here! Please teach me how to fix it?

I finally downgraded to V1... btw: the `useMediaDevices` hook in V1 could be better...it was not returning all I needed - so,  I created my own hook:

```ts
export const useCameraDevices = () => {
    const {
        data,
        isLoading,
        error
    } = useSWR("detectCameraDevices", async () => {
    
        if(isBrowser){
            const devices = await navigator.mediaDevices.enumerateDevices();
            return devices.filter(({kind}) => kind === "videoinput");
        }
        return []
    })

    return {
        devices: data ?? [],
        isLoading: isLoading,
        error
    }
}
```
> Note that I'm on NextJS 14 with SSR and RSC...and further deps on `swr` lib

**Comment by Tstassin** [comment]
Version 1.2.10 !

We experience "continuous scanning" problems too that happen from time to time on an always on QR Scanner kiosk.
The value returned from the scanner when succesfull is an object akin to :
 
<img width="497" alt="Capture d’écran 2024-03-28 à 13 48 14" src="https://github.com/yudielcurbelo/react-qr-scanner/assets/10681663/5c79b67d-c57d-4a5a-8827-ea18403a5004">

But when the bug happens we get a short string, each second, like :

<img width="269" alt="Capture d’écran 2024-03-28 à 13 49 45" src="https://github.com/yudielcurbelo/react-qr-scanner/assets/10681663/7eaefb00-e169-440c-989f-bb7b6d3dd72e">

The bug is hard to track and replicate, the only time we triggered without it happening randomly was just after a succesful scan

**Comment by Tstassin** [comment]
We can confirm that the bug can happen anytime, after seconds, minutes or hours.
The scanner becomes stuck as if it was scanning the same valid value on repeat and only a refresh of the webpage can reset the state. It's hard to understand how to trigger it, but it seems to be triggered by random movements in front of the camera.

**Comment by jakub-sekula** [comment]
Are there any ideas on how to fix this?

**Comment by Tstassin** [comment]
It could be tied to https://github.com/zxing-js/library which is used as dependency, which btw looks in maintenance mode only and has these issues open
- https://github.com/zxing-js/library/issues/589
- https://github.com/zxing-js/library/issues/575

**Comment by macornwell** [comment]
I don't know if it "fixes" any issues behind the scenes but I've added a timeout, after i've collected a scan result, before I collapse (delete) the component.

It appears to have "fixed" this. (It no longer makes beeping noises, and appears to be off).
```
setTimeout(() => {
      setOpen(false)
    }, 1000)
```

**Comment by JoepdeJong** [comment]
We experienced similar issues. We solved it by enabling the scanner in `useEffect`:

```js
function App() {
  const [enabled, setEnabled] = useState<boolean>(false);

  useEffect(() => {
    setEnabled(true);
  }, []);

  return (
    <LayOut>
      <Scanner
        onResult={(text) => {
          setEnabled(false);
          // DO ACTION
        }}
        onError={(error) => console.error(error)}
        enabled={enabled}
      />
      <div id="scanner-video"></div>
    </LayOut>
  );
}

export default App;
```

**Comment by yudielcurbelo** [comment]
This issue should be fixed on `2.0.0-beta.8`. There are breaking changes for this beta release.
